src/main/java/com/blackducksoftware/integration/jira/workflow/setup/BlackDuckFieldScreenSchemeSetup.java

```diff
@@ -312,27 +312,32 @@ private boolean addBlackDuckTabToScreen(final FieldScreen blackDuckScreen, final
             needToUpdateTabAndScreen = true;
         }
         if (blackDuckCustomFields != null && !blackDuckCustomFields.isEmpty()) {
-            for (final OrderableField field : blackDuckCustomFields) {
+            for (OrderableField field : blackDuckCustomFields) {
                 if (field == null) {
                     logger.error("addBlackDuckTabToScreen(): this Black Duck custom field is null; skipping it");
                     continue;
                 }
-                final FieldScreenLayoutItem existingField = myTab.getFieldScreenLayoutItem(field.getId());
-                if (existingField == null) {
+                // Stream over the layout items to prevent the Jira internal Duplicate Key exception.
+                Optional<FieldScreenLayoutItem> existingField = myTab.getFieldScreenLayoutItems().stream()
+                                                                    .filter(fieldScreenLayoutItem -> fieldScreenLayoutItem.getFieldId().equals(field.getId()))
+                                                                    .findFirst();
+                if (!existingField.isPresent()) {
                     logger.debug("addBlackDuckTabToScreen(): custom field " + field.getName() + " is not on Black Duck screen tab; adding it");
                     myTab.addFieldScreenLayoutItem(field.getId());
                     needToUpdateTabAndScreen = true;
                 }
             }
         }
         if (defaultTabs != null && !defaultTabs.isEmpty()) {
-            for (final FieldScreenTab tab : defaultTabs) {
-                final List<FieldScreenLayoutItem> layoutItems = tab.getFieldScreenLayoutItems();
-                for (final FieldScreenLayoutItem layoutItem : layoutItems) {
-                    FieldScreenLayoutItem existingField = null;
+            for (FieldScreenTab tab : defaultTabs) {
+                List<FieldScreenLayoutItem> layoutItems = tab.getFieldScreenLayoutItems();
+                for (FieldScreenLayoutItem layoutItem : layoutItems) {
                     if (configIsOk(myTab, layoutItem)) {
-                        existingField = myTab.getFieldScreenLayoutItem(layoutItem.getOrderableField().getId());
-                        if (existingField == null) {
+                        // Stream over the layout items to prevent the Jira internal Duplicate Key exception.
+                        Optional<FieldScreenLayoutItem> existingField = myTab.getFieldScreenLayoutItems().stream()
+                                                                            .filter(fieldScreenLayoutItem -> fieldScreenLayoutItem.getFieldId().equals(layoutItem.getOrderableField().getId()))
+                                                                            .findFirst();
```

Comment:
Since it's so important we get the FieldScreenLayoutItem this way, we should make it a private method with a single comment on it. We can replace the other call on 321 as well.